tools/src/main/java/com/nedap/archie/flattener/InMemoryFullArchetypeRepository.java

```diff
@@ -69,4 +69,13 @@ public ArchetypeValidationSettings getArchetypeValidationSettings() {
     public void setArchetypeValidationSettings(ArchetypeValidationSettings settings) {
         this.archetypeValidationSettings = settings;
     }
+
+    @Override
+    public void removeArchetype(String archetypeId) {
+        super.removeArchetype(archetypeId);
+        this.flattenedArchetypes.remove(archetypeId);
+        this.operationalTemplates.remove(archetypeId);
+        this.validationResult.remove(archetypeId);
+    }
```

Comment:
This might make other flattenedArchetypes or operationalTemplates in this repository invalid because a parent might be removed. Is that ok?

Reply:
I solved this in the vscode-extension,by invalidating all child archetypes when this happens. I could copy this code here as well. Technically, even archetypes that reference this archetype should be invalidated - I don't think I made that, but not sure.

I might be able to copy this code here, but that is a lot more complex. I think for this PR just adding a comment is a good idea?

Reply:
added comment

Reply:
Yes, a comment is fine for now